### PR TITLE
support Kustomize projects in `skaffold init`

### PR DESCRIPTION
**Pull request by nkubala**
this change adds support for bootstrapping `kustomize` projects through `skaffold init`. Skaffold will look for the presence of any `kustomization.yaml`, and if found, produce a deploy config that uses `kustomize` instead of `kubectl`.

### CODE REVIEW

**Comment by tstromberg** [comment]
Is this going to get an integration test?

**Comment by nkubala** [comment]
@tstromberg it has one! I added a test that runs `skaffold init` on the kustomize example, and then runs `skaffold dev` on the generated skaffold.yaml to make sure everything works properly.

File: pkg/skaffold/deploy/kustomize.go [hunk 1]
```diff
@@ -363,3 +368,13 @@ func buildCommandArgs(buildArgs []string, kustomizePath string) []string {
 
 	return args
 }
+
+func IsKustomizationPath(path string) bool {
```
Comment (balopat) [inline pkg/skaffold/deploy/kustomize.go:372]: I'm wondering if we would want to add some logic to have at least one `resources:` section in the file to make it a kustomization file?

File: pkg/skaffold/initializer/testdata/init/getting-started-kustomize/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+resources:
+  - deployment.yaml
+patches:
```
Comment (balopat) [inline pkg/skaffold/initializer/testdata/init/getting-started-kustomize/kustomization.yaml:3]: Nit: File lists are deprecated for patches.

File: pkg/skaffold/initializer/deploy/kustomize.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+		images:         images,
+		kustomizations: kustomizations,
+	}
+}
+
+// deployConfig implements the Initializer interface and generates
+// a kustomize deployment config.
+func (k *kustomize) DeployConfig() latest.DeployConfig {
+	var kustomizeConfig *latest.KustomizeDeploy
+	// if we only have the default path, leave the config empty - it's cleaner
```
Comment (balopat) [inline pkg/skaffold/initializer/deploy/kustomize.go:50]: 👍

File: pkg/skaffold/initializer/deploy/kustomize.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+	if len(k.kustomizations) == 1 && k.kustomizations[0] == deploy.DefaultKustomizePath {
+		kustomizeConfig = &latest.KustomizeDeploy{}
+	} else {
+		kustomizeConfig = &latest.KustomizeDeploy{
+			KustomizePaths: k.kustomizations,
```
Comment (balopat) [inline pkg/skaffold/initializer/deploy/kustomize.go:55]: so, I'm a bit hesitant around this - 
typical users might have this setup:

```
├── base
│   ├── deployment.yaml
│   └── kustomization.yaml
├── dev
│   ├── kustomization.yaml
│   └── patch.yaml
├── prod
│   ├── kustomization.yaml
│   └── patch.yaml
├── skaffold-kustomize-args.yaml
└── skaffold.yaml
```

And then we get a skaffold.yaml like this: 

```
apiVersion: skaffold/v2beta1
kind: Config
metadata:
  name: kustomize
deploy:
  kustomize:
    paths:
    - base
    - dev
    - prod
```

So...I would consider doing this instead: 
1. we should skip `base` 
2. for each kustomize file we would generate a profile...

Could be done in a follow-up PR. But again, design doc, user journeys would be great to tackle.

Comment (nkubala) [reply pkg/skaffold/initializer/deploy/kustomize.go:55]: yeah, interesting. I wonder if we would want to prompt the user for their "default" kustomization so they don't always have to run with a profile? I'd imagine if users have a "dev" and a "prod" configuration, they'd clearly want "dev" as the default. this is good feedback though, I'm gonna follow this PR up with some improvements.

File: pkg/skaffold/deploy/kustomize.go (hunk 1 above)
Comment (nkubala) [reply pkg/skaffold/deploy/kustomize.go:374]: yeah not a bad idea

File: pkg/skaffold/initializer/testdata/init/getting-started-kustomize/kustomization.yaml (hunk 2 above)
Comment (nkubala) [reply pkg/skaffold/initializer/testdata/init/getting-started-kustomize/kustomization.yaml:3]: and we still have them in all of our examples :)

**Comment by nkubala** [comment]
thanks for the feedback here, I completely agree having some user journeys to go off here would go a long way. i'm gonna merge this for now, and address your improvements in a follow up PR.

**Pull request merged**